src/main/java/org/jabref/logic/importer/fetcher/CiteSeer.java

```diff
@@ -45,32 +49,70 @@ public Optional<HelpFile> getHelpPage() {
         return Optional.of(HelpFile.FETCHER_CITESEERX);
     }
 
+    @Override
+    public List<BibEntry> performSearch(QueryNode luceneQuery) throws FetcherException {
+        // ADR-0014
+        URL urlForQuery;
+        try {
+            urlForQuery = getURLForQuery(luceneQuery);
+        } catch (URISyntaxException | MalformedURLException | FetcherException e) {
+            throw new FetcherException("Search URI crafted from complex search query is malformed", e);
+        }
+
+        String payload = getPayloadString(new DefaultQueryTransformer().transformLuceneQuery(luceneQuery).orElse(""));
+        return getBibEntries(urlForQuery, payload);
+    }
+
+    private List<BibEntry> getBibEntries(URL urlForQuery, String payload) throws FetcherException {
+        try {
+            JsonNode requestResponse = Unirest.post(API_URL)
+                                                     .header("authority", "citeseerx.ist.psu.edu")
+                                                     .header("accept", "application/json, text/plain, */*")
+                                                     .header("content-type", "application/json;charset=UTF-8")
+                                                     .header("origin", "https://citeseerx.ist.psu.edu")
+                                                     .body(payload)
+                                                     .asJson().getBody();
+
+            Optional<JSONArray> responses = Optional.of(requestResponse)
+                                                    .map(JsonNode::getObject)
+                                                    .filter(Objects::nonNull)
+                                                    .map(jsonResponse -> jsonResponse.optJSONArray("response"))
+                                                    .filter(Objects::nonNull);
+
+            InputStream stream = new ByteArrayInputStream(responses.orElse(new JSONArray()).toString().getBytes(StandardCharsets.UTF_8));
```

Comment:
Please check `if (!responses.isPresent()) { return List.of() }`.

Do not convert the JSON to a String to JSON (in the CiteSeerPaser)